### ISSUE TITLE
Recyclable Cache Keys?

### ISSUE THREAD

**Issue by justin808** [DmitryTsepelev/graphql-ruby-fragment_cache#66]
Does this gem support cache_versioning=true?

```ruby
config.active_record.cache_versioning = true
```

https://github.com/rails/rails/pull/29092
https://bigbinary.com/blog/rails-adds-support-for-recyclable-cache-keys

**Comment by DmitryTsepelev** [comment]
Hi @justin808! The gem does not have a `rails` dependency, and cache store is a black box for us. If `config.active_record.cache_versioning = true` setting takes care of everything then yes, we do support it 🙂

**Comment by justin808** [comment]
Rails.cache is an abstraction layer on top of the low-level cache, so if you don't have a Rails dependency on 

https://github.com/rails/rails/blob/main/activesupport/lib/active_support/cache.rb#L226-L373

Then you're not using it.

2. If you convert the key into a string using join, you prevent the Rails cache store from breaking apart the versioning (like timestamp) from the object id.
https://github.com/DmitryTsepelev/graphql-ruby-fragment_cache/blob/master/lib/graphql/fragment_cache/cache_key_builder.rb#L128-L134

In a nutshell:

1. Need to use ActiveSupport's cache
2. Pass cache key as array, so don't convert it to a string using `join`

**Comment by justin808** [comment]
Ahh! I see: https://github.com/DmitryTsepelev/graphql-ruby-fragment_cache/blob/4e35f6e03863306d526438f4fc84d1b7c730d8d5/lib/graphql/fragment_cache/railtie.rb
